### ISSUE TITLE
Cluster: Master-files are not synched with original linux-permissions

### ISSUE THREAD

**Issue by fugstrolch** [Icinga/icinga2#5376]
We have one master-zone with two Icinga2 Cluster Nodes.
The "1st Master's" (=M1) conf-files are saved under /etc/icinga2/zones.d/ .
The conf-files are synched to the "2nd Master" (=M2) and stored under /var/lib/icinga2/api/zones/ .

We have written our own notification-scripts (bash)  -  stored under /etc/icinga2/zones.d/scripts/ (M1).
To enable Icinga2 to start this scripts we have to change the linux-permissions (chmod 755 or chmod a+rx).

Maybe: zones.d is only made for storing conf-files, but it's a good way to make scripts zonewide accessible too :) .

(sorry for my english)

## Expected Behavior

Notifications should be sent from M1 AND M2 whithout manual changes in M2-ConfDir.
Icinga2 should sync permissions of zones.d-files from M1 to M2.

## Current Behavior

Users are getting notifications from M1.
No notifications are sent from M2.
  M2: /var/log/icinga2:  There are logfile-entries that M2 is not allowed to start the notification-script (permission denied).
  M2: The files under /var/lib/icinga2/api/zones/ are not synched with the chmod-Changes.
 
Icinga2 doesn't overwrite permissions of existing files.
Icinga2 doesn't sync permissions of new files.

## Possible Solution

Workaround: We start an hourly cronjob on M2 with:
   find .../scripts/ -type -f -exec chown a+rw {} \;
That's "ok" for us and so this Issue isn't very urgent.

## Steps to Reproduce (for bugs)

1. if you have a cluster with two Icinga2 nodes:
     M1:
        cd /etc/icinga2/zones.d/masterzone
        echo "test" >testfile ; chmod a+rx testfile ; ls -l testfile
                 -rwxr-xr-x    ...   testfile
        icinga2 daemon -C ; /bin/systemctl reload icinga2.service        
     M2:
       cd /var/lib/icinga2/api/zones/masterzone/_etc ; ls -l testfile
                -rw-r--r--    ...   testfile 

## Your Environment

* Version used (`icinga2 --version`):        version: v2.6.3
* Operating System and version:            CentOS Linux release 7.3.1611 (Core) 
                                                             kernel: 3.10.0-514.21.1.el7.x86_64
* Enabled features (`icinga2 feature list`):  Enabled features: 
                                       api checker command graphite ido-mysql livestatus mainlog notification

* Icinga Web 2 version and modules (System - About):
* Config validation (`icinga2 daemon -C`):
* If you run multiple Icinga 2 instances, the `zones.conf` file (or `icinga2 object list --type Endpoint` and `icinga2 object list --type Zone`) from all affected nodes.

**Comment by dnsmichi** [comment]
> Maybe: zones.d is only made for storing conf-files, but it's a good way to make scripts zonewide accessible too :) .

That's exactly the point here: The cluster **config** sync allows to sync icinga2 configuration files only. Everything else which includes executable bits for scripts and whatnot is not supported.

You might not want to hear it, but tools like Puppet, Ansible, Chef or Salt do a great job for managing additional scripts and plugins. 

Further keep in mind with such workarounds inside zones.d - if we ever change the layout in /var/lib/icinga2/api/zones, your notification commands will just break. 

Long story short, this is something we won't add to Icinga 2.

**Comment by fugstrolch** [comment]
hello Michi,
thanks a lot for your answer. That's ok for me.
So we have to look for another solution. Maybe "rsync" is our friend.
Thanks!

**Comment by dnsmichi** [comment]
You could also go for a central git repository which gets pulled by local scripts via cron. I remember doing that years ago in Vienna, when we did not know about Puppet or similar things.
